python_transport/wirepas_gateway/transport_service.py

```diff
@@ -27,6 +27,127 @@
 IMPLEMENTED_API_VERSION = 1
 
 
+class ConnectionToBackendMonitorThread(Thread):
+
+    # Maximum cost to disable traffic
+    SINK_COST_HIGH = 254
+
+    """
+    Thread monitoring the connection with the broker.
+    Connection status is monitored thanks to the number of
+    messages pushed to the mqtt client queue but not published
+    and the latest successfully published packet
+    Mqtt connection is not used as a trigger as it may modify too often
+    the sink cost with an unstable connection that would be counterproductive
+
+    It could also be implemented without a periodic check and tested
+    each time a packet is published but on performance point of view it would
+    be the same
+    """
+    def __init__(self, logger, period, mqtt_wrapper, sink_manager,
+                 minimum_sink_cost, max_buffered_packets, max_delay_without_publish):
```

Comment:
Could you explain the attributes in the Class's doc string? 

I think the max_buffered_packets should have a minimum value.

Reply:
You mean a maximum value? an upper bound? So if not defined by user it is 0 and right now the algorithm is disabled and messages are buffered forever.
What would be the maximum value?
I think the correct solution would be to define a maximum and start overriding old messages if limit is reached but will not be implemented in this change.
I guess for now, this upper limit can be defined by user.

Reply:
I meant maximum value 😅 got a bit mixed up with the parameter name. I think a few k messages would be good and as you said, starting to pop old messages to make room for new ones.

Fine with me if we tackle this in a future optimization.